src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeCleanupFixer.cs

```diff
@@ -270,45 +278,103 @@ public override Task<bool> FixAsync(ICodeCleanUpScope scope, ICodeCleanUpExecuti
             }
         }
 
-        private Task<bool> FixHierarchyContentAsync(IVsHierarchyCodeCleanupScope hierarchyContent, ICodeCleanUpExecutionContext context, CancellationToken cancellationToken)
+        private async Task<bool> FixHierarchyContentAsync(IVsHierarchyCodeCleanupScope hierarchyContent, ICodeCleanUpExecutionContext context, CancellationToken cancellationToken)
         {
-            // TODO: this one will be implemented later
-            // https://github.com/dotnet/roslyn/issues/30165
             var hierarchy = hierarchyContent.Hierarchy;
             if (hierarchy == null)
             {
-                // solution
-                return Task.FromResult(false);
+                return await FixSolutionAsync(_workspace.CurrentSolution, context, cancellationToken).ConfigureAwait(true);
             }
 
             var itemId = hierarchyContent.ItemId;
             if (itemId == (uint)VSConstants.VSITEMID.Root)
             {
-                // Project
+                // Map the hierarchy to a ProjectId. For hierarchies mapping to multitargeted projects, the first target
```

Comment:
>first target [](start = 108, length = 12)

Is it possible to choose the current target based on the active project context? Tag @jasonmalinowski if there is an easy way to do so, otherwise it is fine to do it as follow-up.

Reply:
Does Solution Explorer have an active project context?